### DISCUSSION TITLE
Question regarding ethnic issues

### DISCUSSION THREAD

**Discussion by leccyril** [exadel-inc/CompreFace#824] [General]
We wanted to have more informations regarding this problem you mentionned several times in discussions. What is really the problem ? is it with compreface SDK or with face recognition algorithms in general ? how can fix the issue for example if we choose your SDK and contribute to improve it, how can we be sure that the accuracy will be good with all etnic people ?

Thank you !

**Comment by pospielov** [comment]
I believe the problem is with the original dataset was used to train the model.
Most of the datasets were built based on public photos, mostly celebrity photos. 
As a result, those datasets contained mostly photos of caucasians. 
In the last couple of years appears new datasets with a good representation of all races. I don't know big enough datasets to train a new model, but they are good enough for testing existing ones. Using those datasets you can check if the resulting model is good with all ethnic people.
Still, I hope in near future, new huge datasets will appear and new open-source models will be published. And we will be able to integrate them into CompreFace

**Comment by leccyril** [comment]
on what branch we have to work if we want to work on java projects ? master is not working on branch 1.0.X ?
________________________________
De : Pospielov Serhii ***@***.***>
Envoyé : mardi 12 juillet 2022 18:44
À : exadel-inc/CompreFace ***@***.***>
Cc : leccyril ***@***.***>; Author ***@***.***>
Objet : Re: [exadel-inc/CompreFace] Question regarding ethnic issues (Discussion #824)


I believe the problem is with the original dataset was used to train the model.
Most of the datasets were built based on public photos, mostly celebrity photos.
As a result, those datasets contained mostly photos of caucasians.
In the last couple of years appears new datasets with a good representation of all races. I don't know big enough datasets to train a new model, but they are good enough for testing existing ones. Using those datasets you can check if the resulting model is good with all ethnic people.
Still, I hope in near future, new huge datasets will appear and new open-source models will be published. And we will be able to integrate them into CompreFace

**Reply by pospielov** [reply]
If you want to contribute to CompreFace, you need to fork the project, then make changes (as this is your fork it can be any branch), and then create a merge request from your fork to CompreFace master branch.

**Comment by leccyril** [comment]
If you want i can contribute in doing evolution of maven dependencies and code fixing...

because i could not build API now and i dont kn ow if it is normal and on what branch i have to work
________________________________
De : Pospielov Serhii ***@***.***>
Envoyé : mardi 12 juillet 2022 18:44
À : exadel-inc/CompreFace ***@***.***>
Cc : leccyril ***@***.***>; Author ***@***.***>
Objet : Re: [exadel-inc/CompreFace] Question regarding ethnic issues (Discussion #824)


I believe the problem is with the original dataset was used to train the model.
Most of the datasets were built based on public photos, mostly celebrity photos.
As a result, those datasets contained mostly photos of caucasians.
In the last couple of years appears new datasets with a good representation of all races. I don't know big enough datasets to train a new model, but they are good enough for testing existing ones. Using those datasets you can check if the resulting model is good with all ethnic people.
Still, I hope in near future, new huge datasets will appear and new open-source models will be published. And we will be able to integrate them into CompreFace
